units/en/unit3/hands-on.mdx

```diff
@@ -0,0 +1,320 @@
+
+```bash
+apt install python-opengl
+apt install ffmpeg
+apt install xvfb
+pip3 install pyvirtualdisplay
+```
+
+```bash
+apt-get install swig cmake freeglut3-dev
+```
+
+```bash
+pip install pyglet==1.5.1
+```
+
+```python
+# Virtual display
+from pyvirtualdisplay import Display
+
+virtual_display = Display(visible=0, size=(1400, 900))
+virtual_display.start()
+```
+
+## Clone RL-Baselines3 Zoo Repo 📚
+You can now directly install from python package `pip install rl_zoo3` but since we want **the full installation with extra environments and dependencies** we're going to clone `RL-Baselines3-Zoo` repository and install from source.
+
+```bash
+git clone https://github.com/DLR-RM/rl-baselines3-zoo
+```
+
+## Install dependencies 🔽
+We can now install the dependencies RL-Baselines3 Zoo needs (this can take 5min ⏲)
+
+But we'll also install:
+- `huggingface_sb3`: Additional code for Stable-baselines3 to load and upload models from the Hugging Face 🤗 Hub.
+
+```bash
+cd /content/rl-baselines3-zoo/
+```
+
+```bash
+pip install -r requirements.txt
+```
+
+```bash
+pip install huggingface_sb3
+```
+
+## Train our Deep Q-Learning Agent to Play Space Invaders 👾
+
+To train an agent with RL-Baselines3-Zoo, we just need to do two things:
+1. We define the hyperparameters in `rl-baselines3-zoo/hyperparams/dqn.yml`
+
+<img src="https://huggingface.co/datasets/huggingface-deep-rl-course/course-images/resolve/main/en/notebooks/unit3/hyperparameters.png" alt="DQN Hyperparameters">
+
+
+Here we see that:
+- We use the `Atari Wrapper` that preprocess the input (Frame reduction ,grayscale, stack 4 frames)
+- We use `CnnPolicy`, since we use Convolutional layers to process the frames
+- We train it for 10 million `n_timesteps`
+- Memory (Experience Replay) size is 100000, aka the amount of experience steps you saved to train again your agent with.
+
+💡 My advice is to **reduce the training timesteps to 1M,** which will take about 90 minutes on a P100. `!nvidia-smi` will tell you what GPU you're using. At 10 million steps, this will take about 9 hours, which could likely result in Colab timing out. I recommend running this on your local computer (or somewhere else). Just click on: `File>Download`.
+
+In terms of hyperparameters optimization, my advice is to focus on these 3 hyperparameters:
+- `learning_rate`
+- `buffer_size (Experience Memory size)`
+- `batch_size`
+
+As a good practice, you need to **check the documentation to understand what each hyperparameters does**: https://stable-baselines3.readthedocs.io/en/master/modules/dqn.html#parameters
+
+
+
+2. We run `train.py` and save the models on `logs` folder 📁
+
+```bash
+python train.py --algo ________ --env SpaceInvadersNoFrameskip-v4  -f _________
+```
+
+#### Solution
+
+```bash
+python train.py --algo dqn  --env SpaceInvadersNoFrameskip-v4 -f logs/
+```
+
+## Let's evaluate our agent 👀
+- RL-Baselines3-Zoo provides `enjoy.py` to evaluate our agent.
```

Comment:
Maybe add a sentence explaining what `enjoy.py` is. It's not super clear from this point

Reply:
✔️